src/components/Category/CategoryOption.tsx

```diff
@@ -1,139 +1,87 @@
-import Grant from '../Icon/ProposalCategories/Grant'
-import Tender from '../Icon/ProposalCategories/Tender'
 import SubItem from '../Icon/SubItem'
 import { ProjectTypeFilter } from '../Search/CategoryFilter'
 
-import { categoryIcons } from './CategoryBanner'
 import './CategoryOption.css'
 
-type Props = Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'children'> & {
+type Props = {
   active?: boolean
-  type: string
+  type: ProposalType | 'all_proposals' | ProjectTypeFilter | 'governance_process'
   count?: number
-  subtypes?: SubtypeOptions[]
   onClick?: () => void
+  href?: string
+  className?: string
+  subcategories?: SubtypeOptions[] | GovernanceProcessType[] | BiddingProcessType[]
+  isSubcategoryActive?: (subcategory: string) => boolean
+  subcategoryHref?: (href: string | undefined, subcategory: string) => string
+  icon: React.ReactElement
+  title: string
 }
 
-const icons: Record<string, any> = {
-  all_proposals: All,
-  all_projects: All,
-  community: Grant,
-  content_creator: Grant,
-  gaming: Grant,
-  platform_contributor: Grant,
-  grants: Grant,
-  bidding_and_tendering: Tender,
-  legacy: Grant,
-  ...categoryIcons,
-}
-
-export const getCategoryIcon = (type: string, variant?: CategoryIconVariant, size?: number) => {
-  const newGrants = Object.values(NewGrantCategory)
-  const newGrantIndex = newGrants.map(toSnakeCase).indexOf(type)
-  const isNewGrant = newGrantIndex !== -1
-  if (isNewGrant) {
-    const icon = getNewGrantsCategoryIcon(newGrants[newGrantIndex])
-    return icon({ variant: variant || CategoryIconVariant.Filled, size: size })
-  }
-
-  const Icon = icons[type]
-
-  return (
-    <div className="CategoryOption__IconContainer">
-      <Icon size="24" />
-    </div>
-  )
-}
-
-const getHref = (href: string | undefined, subtype: SubtypeOptions) => {
-  const url = new URL(href || '/', 'http://localhost') // Create a URL object using a dummy URL
-  if (subtype === SubtypeAlternativeOptions.All) {
-    url.searchParams.delete('subtype')
-  } else {
-    url.searchParams.set('subtype', subtype)
-  }
-  const newHref = url.pathname + '?' + url.searchParams.toString()
-  return newHref
-}
-
-export default function CategoryOption({ active, type, className, count, subtypes, onClick, ...props }: Props) {
+export default function CategoryOption({
+  href,
+  active,
+  type,
+  className,
+  count,
+  onClick,
+  icon,
+  title,
+  subcategories,
+  subcategoryHref,
+  isSubcategoryActive,
+}: Props) {
   const t = useFormatMessage()
   const params = useURLSearchParams()
   const currentType = useMemo(() => params.get('type'), [params])
-  const currentSubtype = useMemo(() => toGrantSubtype(params.get('subtype')), [params])
-
-  const handleClick = useCallback(
-    (e: React.MouseEvent<HTMLAnchorElement>) => {
-      if (onClick) {
-        onClick(e)
-      }
-    },
-    [onClick]
+  const isGroupSelected = useMemo(
+    () => !!subcategories?.includes(currentType as never) || currentType === type,
```

Comment:
WDYT about defining a type for subcategories?

Like:

```typescript
type CategoryOptionSubcategories = SubtypeOptions | GovernanceProcessType | BiddingProcessType

type Props = {
  ...
  subcategories?: CategoryOptionSubcategories[]
...
}

export default function CategoryOption({
  ...
}: Props) {
...
  const isGroupSelected = useMemo(
    () => (!!currentType && !!subcategories?.includes(currentType as CategoryOptionSubcategories)) || currentType === type,
    [subcategories, currentType, type]
  )
  ```

Reply:
with this 

```typescript
type CategoryOptionSubcategories = SubtypeOptions | GovernanceProcessType | BiddingProcessType

type Props = {
  ...
  subcategories?: CategoryOptionSubcategories[]
...
}
```

It would be allowed to give an array like this: 
```typescript
[NewGrantCategory.Accelerator, ProposalType.Poll]
```
I don't find it a problem in this particular case to cast to never